pkg/deploy/controller/config_change_controller.go

```diff
@@ -17,27 +19,18 @@ import (
 type DeploymentConfigChangeController struct {
 	ChangeStrategy       ChangeStrategy
 	NextDeploymentConfig func() *deployapi.DeploymentConfig
-	DeploymentStore      cache.Store
 	Codec                runtime.Codec
 	// Stop is an optional channel that controls when the controller exits
 	Stop <-chan struct{}
 }
 
-// ChangeStrategy knows how to generate and update DeploymentConfigs.
-type ChangeStrategy interface {
-	GenerateDeploymentConfig(namespace, name string) (*deployapi.DeploymentConfig, error)
-	UpdateDeploymentConfig(namespace string, config *deployapi.DeploymentConfig) (*deployapi.DeploymentConfig, error)
-}
-
 // Run watches for config change events.
 func (dc *DeploymentConfigChangeController) Run() {
-	go util.Until(func() { dc.HandleDeploymentConfig() }, 0, dc.Stop)
+	go util.Until(func() { dc.HandleDeploymentConfig(dc.NextDeploymentConfig()) }, 0, dc.Stop)
 }
 
 // HandleDeploymentConfig handles the next DeploymentConfig change that happens.
-func (dc *DeploymentConfigChangeController) HandleDeploymentConfig() {
-	config := dc.NextDeploymentConfig()
-
+func (dc *DeploymentConfigChangeController) HandleDeploymentConfig(config *deployapi.DeploymentConfig) error {
```

Comment:
It's already "HandleSomething", and one of the controller Handles more than one thing. Doesn't seem worth changing to me.